### ISSUE TITLE
hwmonitor@sylfurd: Needs changes for vertical panel to work correctly

### ISSUE THREAD

**Issue by attila123** [linuxmint/cinnamon-spices-applets#1708]
```
 * Applet version/Build date: 2017-Nov-20 (basically the same as in git)
 * Cinnamon version (cinnamon --version): 3.6.7
 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...): AntergOS / does not matter
 * Graphics hardware *and* driver used: Not sure / does not matter
 * Snippet or attachment of ~/.cinnamon/glass.log
 * Snippet or attachment of ~/.xsession-errors
 ```

**Notify author of applet**
Applet has no issued author/maintainer as per https://github.com/linuxmint/cinnamon-spices-applets/blob/master/.github/CODEOWNERS @brownsr ?

**Issue**
The applet was designed to horizontal panel. It does not work well on vertical panel. Only the leftmost part of the CPU graph is shown (which gives a delayed values of CPU usage, so not really useful). I hate horizontal panels. They are simply not practical. Please make it work on vertical panels.

**Steps to reproduce**
Add the applet to a horizontal panel. It works correctly showing the CPU and MEM graphs from left to right.
To make the panel vertical: right click on your panel -> Modify panel -> Move panel. Click it on either the left or right part of the screen (where the possible new places for the panel are shown). Only the CPU graph is shown and I believe it is the leftmost part of it, so it shows delayed data.
For vertical panels it make sense to make them wider, e.g. max out the Panel width in Panel settings.

**Expected behaviour**
This applet should check whether it is running on a horizontal or vertical panel (hopefully there is some notification about that). For a horizontal panel, it would render itself the usual way.
For a vertical panel, it needs to get the current panel width and draw the CPU and MEM graphs in such a width, and put them on top of each other (e.g. the CPU could be on top of MEM). If the width of the panel changes, the applet should redraw itself accordingly (so that the user does not have to remove/re-add it because of that).
Bonus feature: the hight (in vertical panel) and the width (on horizontal panel) should be configurable. Looks a bit tall-ish on vertical panel.

**Other information**
I saw that the workspace switcher applet is aware of the horizontal/vertical panel thing. See /usr/share/cinnamon/applets/workspace-switcher@cinnamon.org/applet.js on filesystem. For the first sight, the required tricks are:
this.setAllowedLayout(Applet.AllowedLayout.BOTH);
Define handlers:
on_orientation_changed
on_panel_height_changed

**Comment by Hultan** [comment]
@attila123 , I just added a scale factor setting to this applet, which might make it somewhat more usable in a vertical panel, when set to the lowest value (1.0). I am not sure when that will be available to you, but try it out when you can.

I like your suggestion for full vertical panel support, by showing the graphs over each other, so I might give that a try. I am assuming that you want the graphs to take up the full width of the vertical panel, but drawn over each other with a fixed height of roughly a "normal" panel height?

**Comment by Hultan** [comment]
I have just made a Pull Request with a working vertical mode. I would appreciate if you tried it out and come back with the result here, so that this issue can be closed.

Ps. you can adjust the height of the graphs in vertical mode in the configuration screen.

**Comment by Hultan** [comment]
@jashix This issue can be closed now, it now works in vertical panels...I don't seem to have permissions to close the issue.

**Comment by claudiux** [comment]
Fixed by #2509.
